### PR TITLE
BUG: Add missing setters  for min and max variable

### PR DESCRIPTION
**Pull request by blowekamp**
ITK standards are for each parameter or ivar to have a separate set
method. The exists methods which set a min/max pair now call the
standard macro based set methods.

### CODE REVIEW

**Review by thewtex — CHANGES_REQUESTED** [review]
Thanks for the patch.

To be consistent with ITK naming convensions and to be more descriptive::

`Min` -> `HistogramMinimum`
`Max` -> `HistogramMaximum`

`MinDistance` -> `MinimumDistance`

**Comment by thewtex** [comment]
The old methods could be removed.

File: include/itkRunLengthTextureFeaturesImageFilter.h [hunk 1]
```diff
@@ -191,6 +193,8 @@ class ITK_TEMPLATE_EXPORT RunLengthTextureFeaturesImageFilter:public ImageToImag
    * Set the min and max (inclusive) pixel value that will be used in
    * generating the histogram.
    */
+  itkSetMacro( MinDistance, RealType);
+  itkSetMacro( MaxDistance, RealType);
```
Comment (blowekamp) [inline include/itkRunLengthTextureFeaturesImageFilter.h]: Rename these to DistanceMinimum and DistanceMaximum too?

Comment (thewtex) [reply include/itkRunLengthTextureFeaturesImageFilter.h]: Yes, `MinimumDistance`, `MaximumDistance`, please.

File: include/itkRunLengthTextureFeaturesImageFilter.h [hunk 2]
```diff
@@ -191,6 +193,8 @@ class ITK_TEMPLATE_EXPORT RunLengthTextureFeaturesImageFilter:public ImageToImag
    * Set the min and max (inclusive) pixel value that will be used in
    * generating the histogram.
```
Comment (blowekamp) [inline include/itkRunLengthTextureFeaturesImageFilter.h]: Documentation of variable needs to be updated.

**Comment by jbvimort** [comment]
This patch looks great! :smiley: 

Therefore, as both the distance and the pixel value are parameters of the histogram (which is a 2D histogram) maybe an even better naming would be: 
- HistogramPixelValueMinimum
- HistogramDistanceMinimum

**Comment by blowekamp** [comment]
I think adding `Histogram` to the distance ivar clarifies its purpose. 👍 

For the GLCM filter I there is no need for the `PixelValue` moniker for the ivar. Then for the RLE filter I am unsure of `PixelValue` should be their for clarity or if consistency with the GLCM would be better....

**Comment by thewtex** [comment]
My understanding is that the parameters are on the same histogram, but for different axes. This suggests an encoding of `<component><axis><value>`, i.e. `HistogramDistanceMinimum` and `HistogramValueMinimum`

**Review by jbvimort — APPROVED** [review]
Everything looks good to me! :+1: 
I built and tested it on my linux just to be sure and everything is fine!

**Comment by jbvimort** [comment]
Maybe it would be helpful to change the version number in setup.py in order to upload new python wheels! :smile:

**Comment by blowekamp** [comment]
Thanks for testing!

I have a couple more changes to do too.

**Comment by blowekamp** [comment]
These changes are standalone, and should be good as I believe all the review comments/request have been addressed.

To clarify the additional changes will be a separate patch related to the Mask Image image, adding typedefs, making a proper optional named input, and then adding another template parameter for the type.

**Review by thewtex — COMMENTED** [review]
Looks good, thanks.

Please update `version` in `setup.py` to `3.0.0` to indicate the API change.

**Comment by blowekamp** [comment]
Does every merge into master upload a package?

If I am planning on making further changes which may effect the API can we hold off on uploading and updating the API version until the changes are complete?

**Comment by thewtex** [comment]
There is no automated package uploading. Yes, if more API changes are anticipated, let's hold off on the version change.

**Review by thewtex — APPROVED** [review]

**Comment by thewtex** [comment]
Travis error is unrelated -- merging.

**Pull request merged**